lib/importer.py

```diff
@@ -852,6 +853,23 @@ def execImport(handle: int, options: dict):
         log(f"cannot retrieve {mediaTypes} items without any library section", xbmc.LOGERROR)
         return
 
+    # Decide if doing fast sync or not, if so set filter string to include updatedAt
+    fastSync = True
+    lastSync = mediaImport.getLastSynced()
+    updatedFilterString = ""
+    watchedFilterString = ""
+
+    # Do we need any other checks for whether we should be doing a fast sync or not? TODO, Jeordy
+    if not lastSync:
+        fastSync = False
+        log("first time syncronizing library, forcing a full syncronization", xbmc.LOGINFO)
+
+    if fastSync:
+        log(f"performing fast syncronization of items viewed or updated since {str(lastSync)}")
+        lastSyncEpoch = parser.parse(lastSync).strftime('%s')
+        updatedFilterString = f"&updatedAt>={lastSyncEpoch}"
+        watchedFilterString = f"&lastViewedAt>={lastSyncEpoch}"
```

Comment:
`lastViewedAt` can be passed into `Section.search()` as a kwargs, see e.g. https://github.com/Montellese/mediaimporter.plex/blob/9bd0b05422d1dd49cc78ba7f2a492dd73a74cc5e/plexapi/library.py#L765
Unfortunately for tvshows it seems to be `show.lastViewedAt`, see https://github.com/Montellese/mediaimporter.plex/blob/9bd0b05422d1dd49cc78ba7f2a492dd73a74cc5e/plexapi/library.py#L826 and for episodes `episode.lastViewedAt`, see https://github.com/Montellese/mediaimporter.plex/blob/9bd0b05422d1dd49cc78ba7f2a492dd73a74cc5e/plexapi/library.py#L829

Concerning `updatedAt` I didn't find any filters but maybe we can ask `python-plexapi` to add them as well then we don't need any extra patches on the library.

Reply:
Looking through the implementation the arguments always get joined with an '=' operation:

https://github.com/JeordyR/mediaimporter.plex/blob/improve/plex_delta_sync/plexapi/utils.py#L103

So we would end up with lastViewedAt=<time> instead of lastViewedAt>=<time>

I didn't see anywhere that their current implementation allows specifying the operator.

Reply:
They already seem to support `!=` as an operator. The approach seems to be that you e.g. specify `{ "country!": "Italy" }` i.e. you add the first part of the operator to the end of the filter name / category. At least that what it looks like when I look at https://github.com/Montellese/mediaimporter.plex/blob/9bd0b05422d1dd49cc78ba7f2a492dd73a74cc5e/plexapi/library.py#L628-L630
Maybe that approach can be extended with `<` and `>` as well. Then we only need to have `updatedAt` as a valid filter and we should be good to go.

BTW while looking through the filters I also saw `addedAt`. What is the value of `updatedAt` for newly added items? Is it the same value `addedAt` or is it empty? If the latter applies we would also have to retrieve based on `addedAt`.

Reply:
Ah I see, I didn't process that at all since variable names can't have operators. I'm guessing it ends up working since it stays in kwargs. Will probly have either pass an unpacked dict or a kwargs dict into it, not standard but should probly work to still fill **kwargs with the contents of kwargs instead of kwargs['kwargs'] = {}, I'll play around with it.

updatedAt ends up populated with almost the same value as addedAt. In Plex when you add a new item, addedAt is populated without updatedAt but it immediately performs an update on the item and updatedAt get populated. It is possible for us to be running an import within that few-second window between add and update and miss an item but I'm not sure the chance of it happening is high enough to warrant an additional API call, what do you think?

Reply:
Got it working with some patches to plexapi. Passing an unpacked dict works, kwargs= does not.

Reply:
Let's go with what you got and if we notice any issues we can still add the call for `addedAt`.